New modules to block lots/serial numbers

clonedagain opened the pull request:
This PR serves the feature request in #110 by pulling in two modules from OdooMRP.

alfredoavanzosc committed:
[NEW mrp_lock_lot]

alfredoavanzosc committed:
[FIX mrp_lock_lot]

alfredoavanzosc committed:
[FIX mrp_lock_lot]

alfredoavanzosc committed:
[FIX mrp_lock_lot]

alfredoavanzosc committed:
[FIX mrp_lock_lot]

oihane committed:
[MOD] Improving lots, added chatter and groups

oihane committed:
[MOD] Translations updated

pedrobaeza committed:
[IMP] Expand AUTHORS

alfredoavanzosc committed:
[IMP] mrp_lock_lot: Remove field "Create in locked status when create a lot" from Configuration - Configuration
  - Warehouse, and put it in product category.

alfredoavanzosc committed:
[FIX] mrp_lock_lot: travis correct errors.

alfredoavanzosc committed:
[FIX] mrp_lock_lot: __openrp__.py correct file.

alfredoavanzosc committed:
[FIX] mrp_lock_lot: Change literal and relocating lot traceability.

oihane committed:
[IMP] Fixed email address

Commit:
[I18N] French translation for mrp_lock_lot

Commit:
[IMP] improved strings in mrp_lock_lot

'Block' should be clearer than 'locked'.
The standard reads 'Serial Number/lot' instead of 'lot' since v7.
Removed unimplemented features from README.

Commit:
[PEP8] fix changes related to new strings

Commit:
[ADD] stock_block_lot

Module to block serial numbers/lots without installing MRP. Split from mrp_lock_lot

Commit:
[ADD] stock_block_lot

Module to block serial numbers/lots without installing MRP. Split from mrp_lock_lot

Commit:
[FIX] auto_install metadata in mrp_lock_lot

Commit:
Merge branch '8.0-block-lot-without-mrp' into 8.0-lock_lot_children_categ

pedrobaeza committed:
[FIX] Added missing descriptions

lbellier committed:
[FIX]: stock_lock_lot

- FIX :  Raise exception if domain is none on quants_get .

sysadminmatmoz committed:
Slovene translations of odoomrp modules

Commit:
Add 'stock_lock_lot/' from commit '0674164d7616299fbb671006ab00562ede39b284'

git-subtree-dir: stock_lock_lot
git-subtree-mainline: 6fc1ac4d274ec7328a3127fe75da74e2ff7b23ab
git-subtree-split: 0674164d7616299fbb671006ab00562ede39b284

Commit:
Add 'mrp_lock_lot/' from commit '5052e9416755a34a79a2a866b8daccc92404e9b7'

git-subtree-dir: mrp_lock_lot
git-subtree-mainline: 952fed0c641c3243baf8c37d733561f1e8117dbe
git-subtree-split: 5052e9416755a34a79a2a866b8daccc92404e9b7

Commit:
*lock_lot : README conforms to OCA guidelines

Fixes https://github.com/OCA/stock-logistics-workflow/issues/110

clonedagain commented:
@pedrobaeza can you check this proposal to move modules from OdooMRP to OCA please?

eLBati commented:
Thanks @clonedagain , please check commits not linked to a github account, like https://github.com/numerigraphe/stock-logistics-workflow/commit/0c0d6bca27b55f944983d952e47213201d7f6ac1

clonedagain commented:
@eLBati Those are commits from OdooMRP. Do you think we can do something about it @pedrobaeza?

pedrobaeza commented:
Yeah, there's a ICLA there, but there's one employee that it's not attached to it. Let me check it and come back.

eLBati commented:
@clonedagain  in the example  of 0c0d6bca27b55f944983d952e47213201d7f6ac1 I think the commit should be modified, as author is `alfredo <alfredo@alfredo-Aspire-5738.(none)>`
See http://stackoverflow.com/a/3042512/1187150 about how to do

clonedagain commented:
OK what is alfredo's email please?

pedrobaeza commented:
User: https://github.com/alfredoavanzosc - E-mail: alfredodelafuente@avanzosc.es

clonedagain commented:
I've fixed the history, please mark "needs review".
(for the records, I followed the instructions at https://help.github.com/articles/changing-author-info/)

pedrobaeza commented on mrp_lock_lot/README.rst:
```diff
@@ -0,0 +1,53 @@
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/stock-logistics-workflow/8.0
```
it should be the runbot branch id, not the name (stock-logistics-workflow)

pedrobaeza commented on mrp_lock_lot/__openerp__.py:
```diff
@@ -0,0 +1,42 @@
+{
+    "name": "MRP Lock Lot",
+    "Summary": "Restrict blocked lots in Manufacturing Orders",
+    "version": "8.0.2.0",
```
Use 8.0.2.0.0

pedrobaeza commented on stock_lock_lot/README.rst:
```diff
@@ -0,0 +1,67 @@
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/stock-logistics-workflow/8.0
```
The same

pedrobaeza commented on stock_lock_lot/__openerp__.py:
```diff
@@ -0,0 +1,45 @@
+{
+    "name": "Stock Lock Lot",
+    "Summary": "Restrict blocked lots in Stock Moves and reservations",
+    "version": "8.0.1.0",
```
8.0.1.0.0

Commit:
[IMP] Module version conforms to OCA std

Commit:
[FIX] runbot url

gurneyalex commented on stock_lock_lot/models/stock_production_lot.py:
```diff
@@ -0,0 +1,77 @@
+    def button_unlock(self):
+        return self.write({'locked': False})
+
+    @api.model
```
you need to also decorate `create` with `api.return(lambda rec: rec.id)` otherwise you break inheriting from your model from the old api.

clonedagain replied:
OK I didn't know about api.return(lambda rec: rec.id), do you have docs or references about that, and maybe an example?

gurneyalex commented on stock_lock_lot/models/stock_production_lot.py:
```diff
@@ -0,0 +1,77 @@
+        return super(StockProductionLot, self).create(vals)
+
+    @api.one
+    def write(self, values):
```
you cannot decorate write with `api.one` : this method returns a boolean and the decorator will wrap this in a list. :bug:

clonedagain replied:
OK I didn't know that, do you have docs or references about that, and maybe an example?

clonedagain replied:
Sorry i'm replying to the wrong message, forget it.

gurneyalex commented on stock_lock_lot/models/stock_quant.py:
```diff
@@ -0,0 +1,24 @@
+
+    def quants_get(self, cr, uid, location, product, qty, domain=None,
+                   restrict_lot_id=False, restrict_partner_id=False,
+                   context=None):
```
is there a good reason for using the old API here?

clonedagain replied:
I suppose that was to ensure compatibility with the v7-API method in the standard.
Can we write it another way?

gurneyalex commented:
review found a couple of issues.

clonedagain commented:
@gurneyalex and others, I tried to fix the issues raised in the review, please let me know if it's OK.

Commit:
[FIX] fixes suggested during review

Revert create to old API: using @api.model alone is not compatible with old API, and even with `@api.returns('stock.production.lot')` or `@api.returns(lambda rec: rec.id)` it does not let the web client get the new record's id whe a lot is created in a many2one popup.
remove api.one from write method
docstrings

clonedagain commented:
@gurneyalex I tried to decorate `create` with `@returns` as suggested but the client failed to get the new record IDs. I'm reverting `create` to old API. Any suggestion?

Commit:
Update file headers WRT OCA convention+add missing contributor

Commit:
Add tests

clonedagain commented:
I'd like to see this merged, so I can add more features to it. @pedrobaeza @alfredoavanzosc @oihane @lbellier can you review it one more time please?

oihane commented:
LGTM :+1:

lbellier commented:
Not tested but seems good. :+1:

clonedagain commented:
@oihane @pedrobaeza please notice I did a small functional change against OdooMRP: I introduced a group to manage locking at 76304a8b68a551a6315c202a21229df24c023f40

clonedagain commented:
Users also complained that they don't see the "blocked/unblocked" status when they're not in the allowed group.
So I propose to revert this group feature and merge, and then I'll propose another PR. @pedrobaeza would you agree to merge then?

pedrobaeza commented:
OK, please remove that commit and I'll merge.

Commit:
PEP8 nitpicking

clonedagain commented:
Offending commit removed, and a small PEP8 fix added to make travis green.

pedrobaeza commented:
OK, merging! :+1: 

Thanks for the efforts.
